ComponentKit/Core/CKComponentMemoizer.mm

```diff
@@ -14,11 +14,41 @@
 #import "CKComponentSubclass.h"
 #import "CKMacros.h"
 #import "CKInternalHelpers.h"
+#import "CKThreadLocalComponentScope.h"
+#import "CKComponentScopeHandle.h"
 
 #include <map>
 
 static NSString *CKComponentMemoizerThreadKey = @"CKComponentMemoizer";
 
+static bool CKCanUseMemoizedComponent(CKComponent *c, id scopeFrameIdentifier) {
+  CKThreadLocalComponentScope *s = CKThreadLocalComponentScope::currentScope();
+
+  NSCAssert(s != nullptr, @"No thread local scope!");
+
+  if (!s) {
+    return false;
+  }
+
+  auto stackTop = s->stack.top();
+
+  // Look for the equivalent child frame
+  CKComponentScopeFrame *oldScopeFrame = [stackTop.equivalentPreviousFrame scopeFrameForClass:c.class identifier:scopeFrameIdentifier];
```

Comment:
You can just make this `scopeFrameForHandleWithGlobalIdentifier:` then. Don't even need to pass in `c.class`.